src/common/kube-helpers.ts

```diff
@@ -16,13 +16,56 @@ function resolveTilde(filePath: string) {
   return filePath;
 }
 
-export function loadConfig(pathOrContent?: string): KubeConfig {
+export class LoadKubeError extends Error {
+  public type: string;
+}
+
+export class AccessError extends LoadKubeError {
+  type = "AccessError";
+
+  constructor(public pathname: string) {
+    super();
+  }
+
+  toString() {
+    return `Failed loading the kube config. Permission denied.\n\n${this.pathname}`;
+  }
+}
+
+export class ExistError extends LoadKubeError {
+  type = "ExistError";
+
+  constructor(public pathname: string) {
+    super();
+  }
+
+  toString() {
+    return `Failed loading the kube config. No such file.\n\n${this.pathname}`;
+  }
+}
+
+export function loadConfig(pathOrContent: string): KubeConfig {
   const kc = new KubeConfig();
+  const pathVersion = resolveTilde(pathOrContent);
```

Comment:
probably unlikely but doing this here risks wrongly modifying the config for the case this should be loaded from string. Also, for when it is a file, the `path.resolve()` call should be done too, since it does some normalization. Though maybe this should be done in `loadFromFile()`...

Reply:
in JS (and TS) `strings` are immutable. All "methods" on strings that return a string as a result return a new string. We cannot do it in "loadFromFile" because that is a method on `KubeConfig` which is a dependency provided type.

Reply:
I was thinking `loadFromFile()` would normalize it for its own purposes (before it opened it). But you're right, the string should match inside and outside the `loadFromFile()` call.